File: party_planner.py
import sys

print("Content-type: text/html\n")

items = [
    ("Cake", 20), ("Balloons", 21), ("Music System", 10), ("Lights", 5),
    ("Catering Service", 8), ("DJ", 3), ("Photo Booth", 15), ("Tables", 7),
    ("Chairs", 12), ("Drinks", 6), ("Party Hats", 9), ("Streamers", 18),
    ("Invitation Cards", 4), ("Party Games", 2), ("Cleaning Service", 11)
]

input_str = sys.stdin.read().strip()  

if not input_str:
    print("<h2>No input provided.</h2>")
    sys.exit()

try:
    indices = [int(i) for i in input_str.split(",")]
except:
    print("<h2>Invalid input.</h2>")
    sys.exit()

selected_items = []
selected_values = []
for i in indices:
    if 0 <= i < len(items):
        name, value = items[i]
        selected_items.append(name)
        selected_values.append(value)

if not selected_values:
    print("<h2>No valid selections.</h2>")
    sys.exit()

base_code = selected_values[0]
for val in selected_values[1:]:
    base_code &= val

adjusted_code = base_code
if base_code == 0:
    adjusted_code += 5
    message = "Epic Party Incoming!"
elif base_code > 5:
    adjusted_code -= 2
    message = "Let's keep it classy!"
else:
    message = "Chill vibes only!"

print("<html><body>")
print("<h2>Selected Items:</h2><ul>")
for item in selected_items:
    print(f"<li>{item}</li>")
print("</ul>")
print(f"<p><strong>Base Code:</strong> {base_code}</p>")
print(f"<p><strong>Final Code:</strong> {adjusted_code}</p>")
print(f"<p><strong>Message:</strong> {message}</p>")
print("</body></html>")

File: party_planner.php
<!DOCTYPE html>
<html>
<head>
    <title>Digital Party Planner</title>
</head>
<body>
    <h1>Plan Your Party!</h1>

    <form method="post">
        <p>Select items (hold Ctrl/Cmd to select multiple):</p>
        <select name="items[]" multiple size="10">
            <option value="0">Cake</option>
            <option value="1">Balloons</option>
            <option value="2">Music System</option>
            <option value="3">Lights</option>
            <option value="4">Catering Service</option>
            <option value="5">DJ</option>
            <option value="6">Photo Booth</option>
            <option value="7">Tables</option>
            <option value="8">Chairs</option>
            <option value="9">Drinks</option>
            <option value="10">Party Hats</option>
            <option value="11">Streamers</option>
            <option value="12">Invitation Cards</option>
            <option value="13">Party Games</option>
            <option value="14">Cleaning Service</option>
        </select>
        <br><br>
        <input type="submit" value="Submit">
    </form>

<?php
if ($_SERVER["REQUEST_METHOD"] == "POST" && isset($_POST["items"])) {
    $indices = $_POST["items"]; 
    $input = implode(",", $indices); 


    $cmd = "python3 /var/www/html/party_planner.py";

    $descriptorspec = [
        0 => ["pipe", "r"], 
        1 => ["pipe", "w"], 
        2 => ["pipe", "w"]  
    ];

    $process = proc_open($cmd, $descriptorspec, $pipes);

    if (is_resource($process)) {
        fwrite($pipes[0], $input);
        fclose($pipes[0]);

        $output = stream_get_contents($pipes[1]);
        fclose($pipes[1]);

        $error = stream_get_contents($pipes[2]);
        fclose($pipes[2]);

        $return_value = proc_close($process);

        echo "<hr><div>$output</div>";

        if ($error) {
            echo "<pre style='color:red;'>$error</pre>";
        }
    } else {
        echo "<p>Error running Python script.</p>";
    }
}
?>
</body>
</html>
